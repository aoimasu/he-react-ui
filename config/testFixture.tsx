import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
