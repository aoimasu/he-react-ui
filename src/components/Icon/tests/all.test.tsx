import testFixture from '../../../../config/testFixture';
import all from '../fixtures/all.fixture';

testFixture(all);
