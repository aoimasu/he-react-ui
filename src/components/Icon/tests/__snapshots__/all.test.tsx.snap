// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> rendered correctly with Add fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Alarm fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Alert fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowLeft fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowLeft fixture 2`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowRight fixture 2`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowUp fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Bookings fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Calendar fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with CaretRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with CheckBoxChecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with CheckBoxUnchecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronLeft fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronUp fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Clock fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Cross fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with CurvedArrow fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Delete fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with DeleteProfile fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with DropDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Edit fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Ellipsis fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Engage fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with HealthEngine fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with HealthEngineWithText fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Help fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Home fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Language fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Logout fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Megaphone fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Patients fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Pencil fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with RadioChecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with RadioUnchecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Search fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Settings fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Sync fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Thumb fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ThumbDark fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ThumbDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with ThumbShadow fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with Tick fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with UserShield fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;

exports[`<Icon /> rendered correctly with View fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <Icon
    name="ArrowUp"
    shadow={false}
  />
</Icon>
`;
